frappe/desk/doctype/form_tour/form_tour.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2021, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.utils import slug
+from frappe.model.document import Document
+
+class FormTour(Document):
+	pass
+
+@frappe.whitelist()
+def get_form_tour_steps(tour_name):
+	def get_step(step_doc):
+		return dict(
+			title=step_doc.title,
+			condition=step_doc.condition,
+			fieldname=step_doc.fieldname,
+			description=step_doc.description,
+			position=slug(step_doc.position),
+		)
+
+	tour = frappe.get_doc('Form Tour', tour_name)
+	return [get_step(d) for d in tour.steps]
```

Comment:
where is this used? Also, `frappe.db.get_doc` can get the required data right?

`doc.steps.forEach`?

Reply:
Yes, `frappe.db.get_doc` and `frappe.tour[doctype] = doc.steps` will work.